src/map/utils/diggingutils.cpp

```diff
@@ -0,0 +1,550 @@
+
+	void UpdateDigDelay(CCharEntity* PChar)
+	{
+	}
+
+/************************************************************************
+*																		*
+*	Return the Chocobo Digging delay									*
+*																		*
+************************************************************************/
+
+	int8 ReturnDigDelay(CCharEntity* PChar)
+	{
+		uint8 delay		= 16;								// Base Delay of 16
+		uint8 skillRank = PChar->RealSkills.rank[58];		// Grab Digging Rank
+		switch(skillRank)
+		{
+			case 9:
+			case 8:
+			case 7:
+			case 6:
+			case 5:
+			case 4:
+			case 3:
+			{
+				delay = 0;
+			}
+			case 2:
+			{
+				delay = 6;
+			}
+			case 1:
+			{
+				delay = 11;
+			}
+		}
+		return delay;
+	}
+
+/************************************************************************
+*																		*
+*	Return the Area Chocobo Digging delay								*
+*																		*
+************************************************************************/
+
+	int8 ReturnAreaDigDelay(CCharEntity* PChar)
+	{
+		uint8 delay		= 60;								// Base Delay of 60
+		uint8 skillRank = PChar->RealSkills.rank[58];		// Grab Digging Rank
+		switch(skillRank)
+		{
+			case 10:
+			{
+				delay = 10;
+			}
+			case 9:
+			{
+				delay = 15;
+			}
+			case 8:
+			{
+				delay = 20;
+			}
+			case 7:
+			{
+				delay = 25;
+			}
+			case 6:
+			{
+				delay = 30;
+			}
+			case 5:
+			{
+				delay = 35;
+			}
+			case 4:
+			{
+				delay = 40;
+			}
+			case 3:
+			{
+				delay = 45;
+			}
+			case 2:
+			{
+				delay = 50;
+			}
+			case 1:
+			{
+				delay = 55;
+			}
+		}
+		return delay;
+	}
+
+/************************************************************************
+*																		*
+*	Calculate the Chocobo Digging skill									*
+*																		*
+************************************************************************/
+
+	int32 CalculateSkillUp(CCharEntity* PChar)
+	{
+		uint8  skillRank = PChar->RealSkills.rank[SKILL_DIG];
+		uint16 maxSkill  = (skillRank + 1) * 100;
+		int32  charSkill = PChar->RealSkills.skill[SKILL_DIG];
+		int8 success = WELL512::irand() % 100 + 1;
+		int32  skillAmount  = 1;
+
+		if (charSkill < maxSkill)
+		{
+			if (success <= 15)
+			{
+				// ShowDebug(CL_CYAN"Digging Skill Increase \n");
+				if((skillAmount + charSkill) > maxSkill)
+				{
+					skillAmount = maxSkill - charSkill;
+				}
+				// Success, give a skill up
+				PChar->RealSkills.skill[SKILL_DIG] += skillAmount;
+				if ((skillRank == 0 && charSkill >= 100) ||
+				(skillRank == 1 && charSkill >= 200) ||
+				(skillRank == 2 && charSkill >= 300) ||
+				(skillRank == 3 && charSkill >= 400) ||
+				(skillRank == 4 && charSkill >= 500) ||
+				(skillRank == 5 && charSkill >= 600) ||
+				(skillRank == 6 && charSkill >= 700) ||
+				(skillRank == 7 && charSkill >= 800) ||
+				(skillRank == 8 && charSkill >= 900) ||
+				(skillRank == 9 && charSkill >= 1000) ||
+				(skillRank == 10 && charSkill >= 1100))
```

Comment:
can probably be shortened to use a for loop from 0 to 10, checking if charSkll >= ((skillRank * 100) + 100) and then increment rank and exit the loop